azurerm/resource_arm_application_gateway.go

```diff
@@ -1925,6 +2039,121 @@ func flattenApplicationGatewayRequestRoutingRules(input *[]network.ApplicationGa
 				}
 			}
 
+			if redirect := props.RedirectConfiguration; redirect != nil {
+				if redirect.ID != nil {
+					redirectId, err := parseAzureResourceID(*redirect.ID)
+					if err != nil {
+						return nil, err
+					}
+					redirectName := redirectId.Path["redirectConfigurations"]
+					output["redirect_configuration_name"] = redirectName
+					output["redirect_configuration_id"] = *redirect.ID
+				}
+			}
+
+			results = append(results, output)
+		}
+	}
+
+	return results, nil
+}
+
+func expandApplicationGatewayRedirectConfigurations(d *schema.ResourceData, gatewayID string) *[]network.ApplicationGatewayRedirectConfiguration {
+
+	vs := d.Get("redirect_configuration").([]interface{})
+	results := make([]network.ApplicationGatewayRedirectConfiguration, 0)
+
+	for _, raw := range vs {
+		v := raw.(map[string]interface{})
+
+		name := v["name"].(string)
+		redirectType := v["redirect_type"].(string)
+
+		targetListenerName := v["target_listener_name"].(string)
+		targetListenerID := fmt.Sprintf("%s/httpListeners/%s", gatewayID, targetListenerName)
+
+		targetUrl := v["target_url"].(string)
+
+		includePath := v["include_path"].(bool)
+		includeQueryString := v["include_query_string"].(bool)
+
+		output := network.ApplicationGatewayRedirectConfiguration{
+			Name: utils.String(name),
+			ApplicationGatewayRedirectConfigurationPropertiesFormat: &network.ApplicationGatewayRedirectConfigurationPropertiesFormat{
+				RedirectType: network.ApplicationGatewayRedirectType(redirectType),
+
+				TargetListener: &network.SubResource{
+					ID: utils.String(targetListenerID),
+				},
+				TargetURL: utils.String(targetUrl),
+
+				IncludePath:        utils.Bool(includePath),
+				IncludeQueryString: utils.Bool(includeQueryString),
+			},
+		}
+
+		results = append(results, output)
+	}
+
+	return &results
+}
+
+func flattenApplicationGatewayRedirectConfigurations(input *[]network.ApplicationGatewayRedirectConfiguration) ([]interface{}, error) {
+	results := make([]interface{}, 0)
+	if input == nil {
+		return results, nil
+	}
+
+	for _, config := range *input {
+		if props := config.ApplicationGatewayRedirectConfigurationPropertiesFormat; props != nil {
+
+			if config.TargetListener == nil && config.TargetURL == nil {
+				return nil, fmt.Errorf("[ERROR] Specify either `target_listener_name` or `target_url`")
+			}
+			if props.TargetListener != nil && config.TargetURL != nil {
+				return nil, fmt.Errorf("[ERROR] Conflict between `target_listener_name` and `target_url` (redirection is either to URL or target listener)")
+			}
+
+			if config.TargetURL != nil && config.IncludePath != nil {
```

Comment:
(same here) can we check if this is `!= nil && != ""`

Reply:
Now done this with utility func